app/src/main/java/app/revanced/manager/ui/component/InstallerStatusDialog.kt

```diff
@@ -0,0 +1,158 @@
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import app.revanced.manager.R
+import app.revanced.manager.ui.viewmodel.PackageInstallerResult
+import com.github.materiiapps.enumutil.FromValue
+
+internal abstract class InstallerStatusDialogModel {
+    abstract var packageInstallerResult: PackageInstallerResult
+    abstract var showInstallerStatusDialog: Boolean
+
+    abstract fun reinstall()
+    abstract fun install()
+}
+
+private typealias InstallerStatusDialogButtonHandler = ((model: InstallerStatusDialogModel) -> Unit)
+private typealias InstallerStatusDialogButton = @Composable (model: InstallerStatusDialogModel) -> Unit
+
+
+@Composable
+internal fun InstallerStatusDialog(
+    model: InstallerStatusDialogModel,
+) {
+    val dialogKind = DialogKind.fromValue(model.packageInstallerResult.status) ?: DialogKind.FAILURE
+
+    AlertDialog(
+        onDismissRequest = {
+            model.showInstallerStatusDialog = false
+        },
+        confirmButton = {
+            dialogKind.confirmButton(model)
+        },
+        dismissButton = {
+            dialogKind.dismissButton?.invoke(model)
+        },
+        icon = {
+            Icon(dialogKind.icon, null)
+        },
+        title = {
+            Text(
+                text = stringResource(dialogKind.title),
+                style = MaterialTheme.typography.headlineSmall.copy(textAlign = TextAlign.Center),
+                color = MaterialTheme.colorScheme.onSurface,
+            )
+        },
+        text = {
+            Column(
+                horizontalAlignment = Alignment.CenterHorizontally,
+                verticalArrangement = Arrangement.spacedBy(16.dp),
+                content = {
+                    Text(stringResource(dialogKind.contentStringResId))
+                }
+            )
+        }
+    )
+}
+
+private fun installerDialogButton(
+    @StringRes buttonStringResId: Int,
+    handler: InstallerStatusDialogButtonHandler = { },
+): InstallerStatusDialogButton = { model ->
+    TextButton(onClick = {
+        model.showInstallerStatusDialog = false
+        handler(model)
+    }) { Text(stringResource(buttonStringResId)) }
+}
+
+// TODO: Change to internal once https://github.com/MateriiApps/enumutil-kt/issues/4 is fixed
+@FromValue("flag")
+enum class DialogKind(
+    internal val flag: Int,
+    internal val title: Int,
+    @StringRes internal val contentStringResId: Int,
+    internal val icon: ImageVector = Icons.Outlined.ErrorOutline,
+    internal val confirmButton: InstallerStatusDialogButton = installerDialogButton(R.string.ok),
+    internal val dismissButton: InstallerStatusDialogButton? = null,
```

Comment:
```suggestion
    val flag: Int,
    val title: Int,
    @StringRes internal val contentStringResId: Int,
    val icon: ImageVector = Icons.Outlined.ErrorOutline,
    val confirmButton: InstallerStatusDialogButton = installerDialogButton(R.string.ok),
    internal val dismissButton: InstallerStatusDialogButton? = null,
```
Is internal necessary?